### PR TITLE
Haroun branch

### PR DESCRIPTION
**Pull request by menaili**

### CODE REVIEW

File: assets/icons/arrow_right.svg
Comment (farhioussama1969) [inline assets/icons/arrow_right.svg:1]: icons file must be like this: [icon_name]_icon.svg

and the same for images: [image_name]_image.png

File: lib/app/core/constants/strings_assets_constants.dart
Comment (farhioussama1969) [inline lib/app/core/constants/strings_assets_constants.dart:1]: user python script to generate assets (Strings, image, icons....)

File: lib/app/modules/hotel_details/views/components/hotel_details_comments.dart
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_comments.dart:1]: all components must be named like this: [component_name]_component.dart

File: lib/app/modules/hotel_details/views/components/hotel_details_comments.dart [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import 'package:rihlatic/app/core/styles/text_styles.dart';
+
+class HotelDetailsComments extends StatelessWidget {
+  const HotelDetailsComments({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      child: Padding(
+        padding: const EdgeInsets.all(8.0),
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_comments.dart]: use: .sp | .w | .h | .r

in sizes

File: lib/app/modules/hotel_details/views/components/hotel_details_comments.dart [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                ),
+              ],
+            ),
+            SizedBox(
+              height: 10.h,
+            ),
+            Text(
+              'Very convenient location just across the road from Paris Nord station...',
+              style: TextStyles.smallBodyTextStyle(context).copyWith(
+                  color: MainColors.textColor(context)!, fontSize: 15.r),
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_comments.dart]: use mediumBodyTextStyle

File: lib/app/modules/hotel_details/views/components/hotel_details_comments.dart [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+              ],
+            )
+          ],
+        ),
+      ),
+      decoration: BoxDecoration(
+          borderRadius: BorderRadius.circular(8),
+          border: Border.all(
+              color: MainColors.textColor(context)!.withOpacity(0.15))),
+      height: 214.5.h,
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_comments.dart]: remove height here to make the component flexible

File: lib/app/modules/hotel_details/views/components/hotel_details_comments.dart [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            ),
+            Text(
+              'Very convenient location just across the road from Paris Nord station...',
+              style: TextStyles.smallBodyTextStyle(context).copyWith(
+                  color: MainColors.textColor(context)!, fontSize: 15.r),
+            ),
+            SizedBox(
+              height: 10.h,
+            ),
+            Row(
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_comments.dart]: error in UI when there are more then 4 pictures, so try to fix this problem with listView or wrap or (show + in last picture)

File: lib/app/modules/hotel_details/views/components/hotel_details_description.dart [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+class HotelDetailsDescription extends StatelessWidget {
+  const HotelDetailsDescription({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 20.0),
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_description.dart]: same comment for size

File: lib/app/modules/hotel_details/views/components/hotel_details_description.dart [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+          child: Row(
+            children: [
+              Expanded(
+                child: Text(
+                  StringsAssetsConstants.descriptionHotelDetails,
+                  style: TextStyles.mediumLabelTextStyle(context).copyWith(
+                    color: MainColors.textColor(context)!,
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_description.dart]: any text style par default have text color, so remote it

File: lib/app/modules/hotel_details/views/components/hotel_details_description.dart [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+                  textAlign: TextAlign.left,
+                ),
+              ),
+            ],
+          ),
+        ),
+        SizedBox(
+          height: 10.h,
+        ),
+        Padding(
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_description.dart]: change this component with this package:

https://pub.dev/packages/readmore

File: lib/app/modules/hotel_details/views/components/hotel_details_description.dart [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+          child: Row(
+            children: [
+              Expanded(
+                child: Text(
+                  'A stay at Dream Downtown places you in the heart of New York, within a 5-minute walk of Read more...',
+                  style: TextStyles.mediumBodyTextStyle(context).copyWith(
+                    color: MainColors.textColor(context)!.withOpacity(0.6),
+                  ),
+                  maxLines: 2,
+                  textAlign: TextAlign.left,
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_description.dart]: remove text align

File: lib/app/modules/hotel_details/views/components/hotel_details_exploring.dart [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 20.0),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(
+            'Exploring the area',
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_exploring.dart]: make sure that all static titles in strings assets json

File: lib/app/modules/hotel_details/views/components/hotel_details_exploring.dart [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+            style: TextStyles.mediumLabelTextStyle(context).copyWith(
+              color: MainColors.textColor(context)!,
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_exploring.dart]: same comment for the color

File: lib/app/modules/hotel_details/views/components/hotel_details_exploring.dart [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+            ),
+            maxLines: 2,
+            textAlign: TextAlign.left,
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_exploring.dart]: same remove it

File: lib/app/modules/hotel_details/views/components/hotel_details_exploring.dart [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+          ),
+          SizedBox(
+            height: 10.h,
+          ),
+          Row(
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_exploring.dart]: change structure of this section like this:

Column 
    Row
       Row Row
    Row
       Row Row

File: lib/app/modules/hotel_details/views/components/hotel_details_exploring.dart [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+                  ),
+                ],
+              ),
+            ],
+          ),
+          SizedBox(
+            height: 10.h,
+          ),
+          TagComponent(
+            width: 392.w,
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_exploring.dart]: remove height
and make width double.infinity

File: lib/app/modules/hotel_details/views/components/hotel_details_policy.dart [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+import 'package:rihlatic/app/core/styles/main_colors.dart';
+import 'package:rihlatic/app/core/styles/text_styles.dart';
+
+class HotelDetailsPolicy extends StatelessWidget {
+  const HotelDetailsPolicy({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 20.0),
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_policy.dart]: same comment for sizes

File: lib/app/modules/hotel_details/views/components/hotel_details_policy.dart [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(
+            'Check-in policy',
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_policy.dart]: same for static labels

File: lib/app/modules/hotel_details/views/components/hotel_details_policy.dart [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+            style: TextStyles.mediumLabelTextStyle(context).copyWith(
+              color: MainColors.textColor(context)!,
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_policy.dart]: color

File: lib/app/modules/hotel_details/views/components/hotel_details_policy.dart [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+            ),
+            maxLines: 2,
+            textAlign: TextAlign.left,
+          ),
+          SizedBox(
+            height: 10.h,
+          ),
+          Text(
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_policy.dart]: put it in expanded then in a row

File: lib/app/modules/hotel_details/views/components/hotel_details_policy.dart [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+            'Very convenient location just across the road from Paris Nord station...',
+            style: TextStyles.mediumBodyTextStyle(context).copyWith(
+              color: MainColors.textColor(context)!.withOpacity(0.6),
+            ),
+          ),
+          SizedBox(
+            height: 10.h,
+          ),
+          TagComponent(
+            width: 392.w,
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_policy.dart]: same for height and width

File: lib/app/modules/hotel_details/views/components/hotel_details_reviews.dart [hunk 19]
```diff
@@ -0,0 +1,245 @@
...
+  const HotelDetailsReviews({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 20),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_reviews.dart]: put it in expanded then in a row

File: lib/app/modules/hotel_details/views/components/hotel_details_reviews.dart [hunk 20]
```diff
@@ -0,0 +1,245 @@
...
+import 'package:rihlatic/app/core/styles/main_colors.dart';
+import 'package:rihlatic/app/core/styles/text_styles.dart';
+
+class HotelDetailsReviews extends StatelessWidget {
+  const HotelDetailsReviews({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 20),
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_reviews.dart]: same com for sz

File: lib/app/modules/hotel_details/views/components/hotel_details_reviews.dart [hunk 21]
```diff
@@ -0,0 +1,245 @@
...
+          ),
+          Row(
+            children: [
+              Text(
+                '4.5',
+                style: TextStyles.mediumLabelTextStyle(context).copyWith(
+                  color: MainColors.secondColor,
+                ),
+                maxLines: 2,
+                textAlign: TextAlign.left,
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_reviews.dart]: remove any text align

File: lib/app/modules/hotel_details/views/components/hotel_details_reviews.dart [hunk 22]
```diff
@@ -0,0 +1,245 @@
...
+          ),
+          SizedBox(
+            height: 10.h,
+          ),
+          Row(
+            children: [
+              Column(
+                children: [
+                  SizedBox(
+                    width: 188.5.w,
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_reviews.dart]: remove  wight

File: lib/app/modules/hotel_details/views/components/hotel_details_reviews.dart [hunk 23]
```diff
@@ -0,0 +1,245 @@
...
+                                  color: MainColors.blackColor.withOpacity(0.6),
+                                )),
+                          ],
+                        ),
+                        SizedBox(
+                          height: 5.h,
+                        ),
+                        LinearPercentIndicator(
+                          padding: EdgeInsets.all(0),
+                          width: 180,
```
Comment (farhioussama1969) [inline lib/app/modules/hotel_details/views/components/hotel_details_reviews.dart]: don't use width just use flexible widget or expanded widget

File: lib/app/core/constants/strings_assets_constants.dart
Comment (menaili) [reply lib/app/core/constants/strings_assets_constants.dart:1]: how, give me exemple...

File: lib/app/modules/hotel_details/views/components/hotel_details_policy.dart (hunk 17 above)
Comment (menaili) [reply lib/app/modules/hotel_details/views/components/hotel_details_policy.dart]: don't understand

Comment (menaili) [reply lib/app/modules/hotel_details/views/components/hotel_details_policy.dart]: done

File: assets/icons/arrow_right.svg
Comment (menaili) [reply assets/icons/arrow_right.svg:1]: done

File: lib/app/modules/hotel_details/views/components/hotel_details_comments.dart
Comment (menaili) [reply lib/app/modules/hotel_details/views/components/hotel_details_comments.dart:1]: done

**Pull request merged**